publication-commons/src/main/java/no/unit/nva/publication/service/impl/RestPublicationService.java

```diff
@@ -0,0 +1,166 @@
+import java.util.List;
+import java.util.UUID;
+
+public class RestPublicationService implements PublicationService {
+
+    public static final String PATH = "/resource/";
+    public static final String ACCEPT = "Accept";
+    public static final String APPLICATION_JSON = "application/json";
+    public static final String AUTHORIZATION = "Authorization";
+    public static final String CONTENT_TYPE = "Content-Type";
+    public static final String API_HOST_ENV = "API_HOST";
+    public static final String API_SCHEME_ENV = "API_SCHEME";
+    public static final String ITEMS_0 = "/Items/0";
+    public static final String ERROR_COMMUNICATING_WITH_REMOTE_SERVICE = "Error communicating with remote service: ";
+    public static final String ERROR_RESPONSE_FROM_REMOTE_SERVICE = "Error response from remote service: ";
+    public static final String ERROR_MAPPING_PUBLICATION_TO_JSON = "Error mapping Publication to JSON";
+    public static final String IDENTIFIERS_NOT_EQUAL = "Identifier in request parameters '%s' "
+            + "is not equal to identifier in customer object '%s'";
+
+    private final ObjectMapper objectMapper;
+    private final HttpClient client;
+    private final String apiScheme;
+    private final String apiHost;
+
+    /**
+     * Constructor for RestPublicationService.
+     *
+     * @param objectMapper objectMapper
+     * @param client    client
+     * @param apiScheme apiScheme
+     * @param apiHost   apiHost
+     */
+    public RestPublicationService(HttpClient client, ObjectMapper objectMapper, String apiScheme, String apiHost) {
+        this.objectMapper = objectMapper;
+        this.client = client;
+        this.apiScheme = apiScheme;
+        this.apiHost = apiHost;
+    }
+
+    /**
+     * Constructor for RestPublicationService.
+     *
+     * @param client    client
+     * @param objectMapper objectMapper
+     * @param environment   environment
+     */
+    public RestPublicationService(HttpClient client, ObjectMapper objectMapper, Environment environment) {
+        this(client, objectMapper, environment.readEnv(API_SCHEME_ENV), environment.readEnv(API_HOST_ENV));
+    }
+
+    @Override
+    public Publication getPublication(UUID identifier, String authorization)
+            throws ApiGatewayException {
+        URI uri = UrlBuilder.empty()
+                .withScheme(apiScheme)
+                .withHost(apiHost)
+                .withPath(PATH + identifier.toString())
+                .toUri();
+
+        HttpRequest httpRequest = HttpRequest.newBuilder()
+                .uri(uri)
+                .header(ACCEPT, APPLICATION_JSON)
+                .header(AUTHORIZATION, authorization)
+                .GET()
+                .build();
+
+        try {
+            HttpResponse<String> httpResponse = client.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+
+            JsonNode jsonNode = objectMapper.readTree(httpResponse.body());
+            JsonNode item0 = jsonNode.at(ITEMS_0);
```

Comment:
What is item0? First publication version/last publication version/latest publication version?